snapshot/snapshot.go

```diff
@@ -77,7 +77,10 @@ const (
 )
 
 var (
-	// Error returned by `fs.Mount` when there is no ztoc for a particular layer.
+	// ErrUnableToLazyLoadImage is returned by `fs.Mount` when the image cannot/should not be
+	// lazy loaded. (eg: If a SOCI index does not exist for an image)
+	ErrUnableToLazyLoadImage = errors.New("unable to lazy load image")
```

Comment:
What about `ErrNoIndex`? I think that `ErrUnableToLazyLoadImage` could be useful somewhere else, but right now you can't distinguish between "there is no index" and "there's an issue with the registry" that might be useful down the line.

Reply:
Made these changes, but this then separates the bad index case from the no index found case. Perhaps we should make a separate error for the former and test that as well?